middleware/kubernetes/kubernetes.go

```diff
@@ -234,7 +248,15 @@ func (k *Kubernetes) parseRequest(lowerCasedName, qtype string) (r recordRequest
 		}
 	}
 	if r.zone == "" {
-		return r, errors.New("zone not found")
+		return r, errZoneNotFound
+	}
+
+	if qtype == "NS" {
```

Comment:
Most dns type comparisons use dns.TypeXX (`dns.TypeNS`), how did we end up with a string here?